app/src/main/java/com/codingblocks/cbonlineapp/course/CourseViewModel.kt

```diff
@@ -221,4 +230,77 @@ class CourseViewModel(
         }
         return LivePagedListBuilder<String, Course>(dataSourceFactory, config)
     }
+
+    fun changeWishlistStatus(id: String? = course.value?.id, mainCourse: Boolean = false){
+        runIO {
+            when (val response = repo.checkIfWishlisted(id?:"")) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        val isWishlisted = response.value.body()?.id!=null
+                        if (mainCourse){
+                            wishlistUpdated.postValue(!isWishlisted)
+                        }
+                        if (isWishlisted){
+                            removeFromWishlist(response.value.body()?.id)
+                        }else{
+                            addToWishlist(id)
+                        }
+                    } else {
+                        setError(fetchError(code()))
+                    }
+                }
+            }
+        }
+    }
+
+    fun checkIfCourseWishlisted(courseSingle: Course?){
+        runIO {
+            when (val response = repo.checkIfWishlisted(courseSingle?.id?:"")) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        wishlistUpdated.postValue(response.value.body()?.id!=null)
+                        course.postValue(courseSingle)
+                    } else {
+                        setError(fetchError(code()))
+                    }
+                }
+            }
+
+        }
+    }
+
+    fun addToWishlist(id: String? = course.value?.id){
```

Comment:
there's no need to pass default argument here and make this function argument as not nullable